Shop_Unity_base/Assets/Scripts/Shop/View/ShopMessageView.cs

```diff
@@ -7,26 +7,33 @@ public class ShopMessageView : MonoBehaviour
     {
         private ShopModel shopModel;
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  Initialize()
-        //------------------------------------------------------------------------------------------------------------------------
-        //This method is used to initialize values, because we can't use a constructor.
+        // This method is used to initialize values, because we can't use a constructor.
         public void Initialize(ShopModel shopModel)
         {
             this.shopModel = shopModel;
+
+            ShopModel.OnBuyEvent += OnShopUpdated;
+            ShopModel.OnSellEvent += OnShopUpdated;
+        }
+
+        private void OnDestroy()
+        {
+            ShopModel.OnBuyEvent -= OnShopUpdated;
+            ShopModel.OnSellEvent -= OnShopUpdated;
         }
 
         //------------------------------------------------------------------------------------------------------------------------
         //                                                  Update()
         //------------------------------------------------------------------------------------------------------------------------        
-        //this method polls the shop for messages and prints them. Since the shop caches the messages, it prints the same
-        //message each frame. An event system would work better.
-        protected void Update() {
+        // This method polls the shop for messages and prints them. Since the shop caches the messages, it prints the same
+        // message each frame. An event system would work better.
+
+        private void OnShopUpdated(Item item, Customer customer)
+        {
             string[] messages = shopModel.GetMessages();
-            if (messages.Length > 0) {
-                string message = messages[messages.Length - 1];
-                Debug.Log(message);
-            }
+
+            for (int i = 0; i < messages.Length; i++)
+                Debug.Log(messages[i]);
```

Comment:
Already outlined in the ShipModel file for my suggestions for the messaging system. I thought the messaging system would not be getting printed but rather Drawn. (as I think it does in the GXP version).